### PR TITLE
feat: add UIConfig and UserUIConfig [EXT-3738]

### PR DESCRIPTION
**Pull request by andipaetzold**
## Summary

Adds new entities UIConfig and UserUIConfig

Integration tests can only be added once the endpoints are available

## TODO

- [x] Remove page field

## Checklist (check all before merging)

- [x] Both unit and integration tests are passing
- [x] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [x] The new method is exported through the default and plain CMA client
- [x] All new public types are exported from `./lib/export-types.ts`
- [x] Added a unit test for the new method
- [ ] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

File: lib/create-ui-config-api.ts
Comment (loweisz) [inline lib/create-ui-config-api.ts:41]: I don't think we need an entry-id here right?

File: lib/entities/user-ui-config.ts
Comment (loweisz) [inline lib/entities/user-ui-config.ts:39]: Can we not share someof the types with the ui-config?

Comment (loweisz) [inline lib/entities/user-ui-config.ts:39]: Can we not share someof the types with the ui-config?

File: lib/create-ui-config-api.ts
Comment (loweisz) [inline lib/create-ui-config-api.ts:41]: also here I think we don't the entry id

Comment (andipaetzold) [reply lib/create-ui-config-api.ts:41]: Copy paste :D

File: lib/entities/user-ui-config.ts
Comment (andipaetzold) [reply lib/entities/user-ui-config.ts:39]: I explicitly kept the types separate in case they diverge from another.
E.g. the `View` has an additional `roles` field

Comment (loweisz) [reply lib/entities/user-ui-config.ts:40]: True, but we could share some parts and extend them for the use case. 
But I'm also fine with two different types 👍

**Review by loweisz — APPROVED** [review]
one typo, then its good to go

File: lib/entities/user-ui-config.ts
Comment (kdamball) [inline lib/entities/user-ui-config.ts:35]: These have to go, no? Or was it the page property in UIConfig itself? @colomolo has a better idea which `page` property was deleted

Comment (kdamball) [reply lib/entities/user-ui-config.ts:35]: Yep, found similar comment from @colomolo  here: https://github.com/contentful/doc-app/pull/2379

Comment (andipaetzold) [reply lib/entities/user-ui-config.ts:35]: I have that in the TODO list in the PR description 🤓

Comment (kdamball) [reply lib/entities/user-ui-config.ts:35]: will wait for approval then

File: lib/entities/user-ui-config.ts
Comment (kdamball) [inline lib/entities/user-ui-config.ts:36]: @colomolo can you confirm this is needed?

Comment (colomolo) [reply lib/entities/user-ui-config.ts:36]: yes, confirming.

**Review by kdamball — APPROVED** [review]

**Review by colomolo — APPROVED** [review]
lgtm :shipit:

**Pull request merged**
